app/uk/gov/hmrc/tai/controllers/JourneyCacheController.scala

```diff
@@ -37,7 +37,7 @@ class JourneyCacheController @Inject()(
   def currentCache(journeyName: String): Action[AnyContent] = authentication.async { implicit request =>
     repository.currentCache(CacheId(request.nino), journeyName) map {
       case Some(cache) if cache.nonEmpty => Ok(Json.toJson(cache))
-      case _                             => NotFound
+      case _                             => NoContent
```

Comment:
Do we need to still return a 404 in some cases, like this? 
```
case Some(cache) if cache.nonEmpty    => Ok(Json.toJson(cache))
case Some(cache) if cache.isEmpty       => NoContent
case _                                                      => NotFound
```

Reply:
i thought the whole idea was to give the same functionality but changing NotFound to NoContent to avoid too many 404s in kibana

Reply:
That wouldn't solve the kibana problem with too many NotFounds.

Reply:
The `repository.currentCache(CacheId(request.nino), journeyName)` recovers errors and turns them to None, maybe we want to fail the futures so the error is propagated as Internal Server Error rather than NoContent in case something's wrong.

cc: @vivekhmrc

Reply:
Hi @sergi-hmrc I have reviewed the PR. For now the changes look correct but I am adding @brendanstraughan 's comment to the Error handling story as a feedback/suggestion